src/task.py

```diff
@@ -0,0 +1,233 @@
+            )
+            task_id = self._parse_task_id(output)
+            if task_id is None:
+                raise TypeError("task_id is None")
+            return task_id
+        except subprocess.CalledProcessError as err:
+            raise ConversionException(err.stdout)
+    
+    def _cp_storage_to_ee_table(self, blob_uri: str, table_asset_id: str) -> str:
+        try:
+            cmd = [
+                "/usr/local/bin/earthengine",
+                f"--service_account_file {self.google_creds_path}",
+                "upload table",
+                "--primary_geometry_column wkt",
+                f"--asset_id={table_asset_id}",
+                blob_uri,
+            ]
+            output = subprocess.check_output(
+                " ".join(cmd), stderr=subprocess.STDOUT, shell=True
+            )
+            task_id = self._parse_task_id(output)
+            if task_id is None:
+                raise TypeError("task_id is None")
+            return task_id
+        except subprocess.CalledProcessError as err:
+            raise ConversionException(err.stdout)
+
+    def _get_image_asset_id(self, attribute: str, tag: str, task_date: str):
+        return f"{self.ee_osm_root}/{attribute}/{tag}/{tag}_{task_date}"
+
+    # Step 1
+    def import_images_to_ee(
+        self, metadata: dict, image_asset_id: Optional[str] = None
+    ) -> List[str]:
+        
+        image_uris = metadata.get("images") or []
+        if not image_uris:
+            return []
+        
+        ee_dir = f"{self.ee_osm_root}/{self.taskdate}"
+        self._prep_asset_id(ee_dir, image_collection=False)
+
+        image_asset_ids = []
+        for image_uri in image_uris:
+            image_asset_id = f"{PROJECTS}/{self.ee_project}/{ee_dir}/{Path(os.path.splitext(image_uri)[0]).name}"
+            task_id = self._cp_storage_to_ee_image(image_uri, image_asset_id)
+            self.ee_tasks[task_id] = {}
+            image_asset_ids.append(image_asset_id)
+
+        self.wait()
+
+        return image_asset_ids
+
+    # Step 2
+    def import_roads_to_ee(
+        self, blob_uri: str, roads_asset_id: Optional[str] = None
+    ) -> str:
+        task_id = self._cp_storage_to_ee_table(blob_uri, roads_asset_id)
+        self.ee_tasks[task_id] = {}
+
+    # Step 3
+    def group_bands(self, image_asset_ids: List[str], metadata: dict) -> str:
+        band_names = [f"{at['attribute']}_{at['tag']}" for at in metadata["bands"].values()]
+        image_stack = ee.ImageCollection.fromImages([ee.Image(i) for i in image_asset_ids])
+        bands = ee.List(list(metadata["bands"].values()))
+        projection = ee.Image(image_asset_ids[0]).projection()
+
+        def band_merge(attr_tag_meta):
+            atm = ee.Dictionary(attr_tag_meta)
+            _bands = ee.List(atm.get("bands"))
+            attribute = atm.get("attribute")
+            tag = atm.get("tag")
+
+            band_indices = _bands.map(lambda x: ee.Number(x).subtract(ee.Number(1)))
```

Comment:
I'm not sure lambdas are serializable and therefore that this behaves as you're intending

Reply:
Never mind -- turns out lambdas are fine. 

In which case this PR looks good to merge to me...